src/core/DataFile.cpp

```diff
@@ -89,18 +89,27 @@ const std::vector<ProjectVersion> DataFile::UPGRADE_VERSIONS = {
 	"1.2.0-rc3"        ,   "1.3.0"
 };
 
-std::array<DataFile::typeDescStruct, DataFile::TypeCount> DataFile::s_types =
+namespace 
 {
-	DataFile::typeDescStruct{ DataFile::UnknownType, "unknown" },
-	DataFile::typeDescStruct{ DataFile::SongProject, "song" },
-	DataFile::typeDescStruct{ DataFile::SongProjectTemplate, "songtemplate" },
-	DataFile::typeDescStruct{ DataFile::InstrumentTrackSettings, "instrumenttracksettings" },
-	DataFile::typeDescStruct{ DataFile::DragNDropData, "dnddata" },
-	DataFile::typeDescStruct{ DataFile::ClipboardData, "clipboard-data" },
-	DataFile::typeDescStruct{ DataFile::JournalData, "journaldata" },
-	DataFile::typeDescStruct{ DataFile::EffectSettings, "effectsettings" },
-	DataFile::typeDescStruct{ DataFile::MidiClip, "midiclip" }
-} ;
+	struct LMMS_EXPORT typeDescStruct
+	{
+		DataFile::Type m_type;
+		QString m_name;
+	};
+
+	auto s_types = std::array<typeDescStruct, DataFile::TypeCount>
```

Comment:
Again, an improvement beyond what this PR was intended for, but this array can be const.